activerecord/lib/active_record/attribute_methods/dirty.rb

```diff
@@ -229,7 +229,13 @@ def attribute_names_for_partial_updates
         end
 
         def attribute_names_for_partial_inserts
-          partial_inserts? ? changed_attribute_names_to_save : attribute_names
+          if partial_inserts?
+            changed_attribute_names_to_save
+          else
+            attribute_names.delete_if do |attr_name|
```

Comment:
This seems wrong, `delete_if` returns what was removed from the array: https://ruby-doc.org/core-2.5.0/Array.html#method-i-delete_if

```ruby
scores = [ 97, 42, 75 ]
scores.delete_if {|score| score < 80 }   #=> [97]
```

So I think you just want `reject` here.

Reply:
@byroot thank you for the review.

I think `delete_if` returns the elements that remained after deletion. In the example code block the return value is `[97]` and deleted items are `[42, 75]` (score < 80).

Reply:
🤦 my bad I should have finished my coffee before reviewing.

Reply:
That said, `reject` would be better, because here you mutate an array, but don't do anything with it.